server/services/yeelight/lib/device/discover.js

```diff
@@ -0,0 +1,68 @@
+const Promise = require('bluebird');
+const logger = require('../../../../utils/logger');
+const features = require('../features');
+const { DEVICE_EXTERNAL_ID_BASE } = require('../utils/constants');
+
+/**
+ * @description Send a broadcast to find the devices
+ * @returns {Promise<Array>} Resolve with array of new devices.
+ * @example
+ * discover();
+ */
+async function discover() {
+  return new Promise((resolve, reject) => {
+    if (!this.discoveryInProgress) {
+      logger.debug(`Yeelight: Discovery in progress...`);
+      const unknownDevices = [];
+      this.discoveryInProgress = true;
+
+      if (!this.discovery || this.discovery.isDestroyed) {
+        this.discovery = new this.yeelightApi.Discover({ fallback: false });
+      }
+
+      // On device found...
+      this.discovery.on('deviceAdded', (discoveredDevice) => {
+        // ...check if it is already in Gladys...
+        const deviceInGladys = this.gladys.stateManager.get(
+          'deviceByExternalId',
+          `${DEVICE_EXTERNAL_ID_BASE}:${discoveredDevice.id}`,
+        );
+        if (deviceInGladys) {
+          logger.debug(`Yeelight: Device "${discoveredDevice.id}" is already in Gladys !`);
+        } else {
+          logger.debug(`Yeelight: Device "${discoveredDevice.id}" found, model: "${discoveredDevice.model}"`);
+          unknownDevices.push(features.getDevice(this.serviceId, discoveredDevice));
+        }
+      });
+
+      this.discovery.on('error', (error) => reject(error));
+
+      this.discovery
+        .start()
+        .then(() => {
```

Comment:
I would recommend using async/await instead of this pattern, we try to be 100% on async/await in Gladys :)